src/main/java/recipeapplication/database/RecipeFirebaseCrud.java

```diff
@@ -22,21 +26,73 @@ public RecipeFirebaseCrud(Firestore firestore,
 
 	@Override
 	public void create(Recipe data) {
-		
-		this.firestore.collection(collection).document(data.getId()).set(data);
-		
+		System.out.println("In recipe creation section");
+
+		try {
+			if (this.firestore.collection(collection).document(data.getId()).get().get().exists() == false)  // Check if document already exists, to not allow overwriting it with create.
+				this.firestore.collection(collection).document(data.getId()).set(data);
+			else
+				throw new Exception("Failed to create, Recipe with this ID already exists!");
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		} catch (ExecutionException e) {
+			e.printStackTrace();
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+		}
 	}
 
 	@Override
-	public void read(String document) {
-		// TODO Auto-generated method stub
-		
+	public Recipe read(String document) {
+		System.out.println("In recipe reading section");
+		try {
+			DocumentSnapshot docSnap = this.firestore.collection(collection).document(document).get().get();
+			if (docSnap.exists() == false)
+				throw new Exception("Failed to read, recipe doesn't exist!");
+			else {
+				Map<String, Object> data = this.firestore.collection(collection).document(document).get().get().getData();
+
+				Recipe recipe = new Recipe();
+				recipe.setId(data.get("id").toString());
+				recipe.setCategory(data.get("category").toString());
+				recipe.setIngridiant(((List<String>) data.get("ingridiant")));
+				recipe.setPreperation(data.get("preperation").toString());
+				if (data.get("recipeImageId") != null)
+					recipe.setRecipeImageId(data.get("recipeImageId").toString());
+				return recipe;
+			}
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+			return null;
+		} catch (ExecutionException e) {
```

Comment:
its not. good behavior to return null as a response, lets throw a runtime exception if something is wrong. This way it will be easier for the client to know that something went wrong

Reply:
Changed to throw runtime exceptions.

Reply:
@SagivAsraf wrote an exception handler for the User Controller, check that out and make a similar exception handler for the Recipe Controller